GlucoseOnFhir/BGMFhir.swift

```diff
@@ -0,0 +1,640 @@
+                    extensionElementCoding.display = FHIRString.init((String(describing:measurement.context!.medicationID!.description)))
+                    
+                    let extensionElement = Extension()
+                    extensionElement.url = FHIRURL.init(bluetoothGlucoseMeasurementContextURL)
+                    extensionElement.valueCoding = extensionElementCoding
+                    
+                    observationExtensionArray.append(extensionElement)
+                }
+                
+                // Medication
+                if measurement.context?.medication?.description != nil {
+                    let extensionElementCoding = Coding()
+                    extensionElementCoding.system = FHIRURL.init(bluetoothGlucoseMeasurementContextURL)
+                    extensionElementCoding.code = FHIRString.init((String(describing:measurement.context!.medication!.description)))
+                    extensionElementCoding.display = FHIRString.init((String(describing:measurement.context!.medication!.description)))
+                    
+                    let extensionElement = Extension()
+                    extensionElement.url = FHIRURL.init(bluetoothGlucoseMeasurementContextURL)
+                    extensionElement.valueCoding = extensionElementCoding
+                    
+                    observationExtensionArray.append(extensionElement)
+                }
+                
+                // hbA1c
+                if measurement.context?.hbA1c?.description != nil {
+                    let extensionElementCoding = Coding()
+                    extensionElementCoding.system = FHIRURL.init(bluetoothGlucoseMeasurementContextURL)
+                    extensionElementCoding.code = FHIRString.init((String(describing:measurement.context!.hbA1c!.description)))
+                    extensionElementCoding.display = FHIRString.init((String(describing:measurement.context!.hbA1c!.description)))
+                    
+                    let extensionElement = Extension()
+                    extensionElement.url = FHIRURL.init(bluetoothGlucoseMeasurementContextURL)
+                    extensionElement.valueCoding = extensionElementCoding
+                    
+                    observationExtensionArray.append(extensionElement)
+                }
+            }
+            observation.extension_fhir = observationExtensionArray
+        }
+        return observation
+    }
+    
+    public func createSpecimen() {
+        let specimen = Specimen()
+        let specimenCollection = SpecimenCollection()
+        
+        let bodySiteCoding = Coding()
+        bodySiteCoding.system = FHIRURL.init("https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.glucose_measurement.xml")
+        bodySiteCoding.code = FHIRString.init(String(describing:"1"))
+        bodySiteCoding.display = FHIRString.init(String(describing:"Finger"))
```

Comment:
I think creating an enum for this coding system would be cleaner. e.g. 
```let site = BluetoothBodySiteCoding.finger
let coding = site.codableConcept
```
At least it hides the ugliness.